x/coin.go

```diff
@@ -172,18 +172,18 @@ func (c *Coin) Clone() *Coin {
 // logic
 func (c Coin) Validate() error {
 	if !IsCC(c.Ticker) {
-		return ErrInvalidCurrency(c.Ticker)
+		return InvalidCurrencyErr.New(c.Ticker)
 	}
 	if c.Whole < MinInt || c.Whole > MaxInt {
-		return ErrOutOfRange(c)
+		return InvalidCoinErr.New(outOfRangeErr)
```

Comment:
req: can you make `InvalidCoinErr.New(outOfRangeErr)` a first class error `var` please instead of a  `const` for the message? I found it very confusing when I first saw it.

Reply:
@alpe there is a reason I did this, reason being we are using the same code with several messages and some of them repeat in several places.
So it did make sense to reuse the string.

I can rename it if it makes things easier, but let's talk about it after the refactor